subsys/usb/device/class/cdc_acm.c

```diff
@@ -56,15 +56,17 @@
 /* definitions */
 
 #include <zephyr/logging/log.h>
-#if DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_console), zephyr_cdc_acm_uart) \
-	&& defined(CONFIG_LOG_BACKEND_UART) \
-	&& defined(CONFIG_USB_CDC_ACM_LOG_LEVEL) \
-	&& CONFIG_USB_CDC_ACM_LOG_LEVEL != LOG_LEVEL_NONE
 /* Prevent endless recursive logging loop and warn user about it */
+#if defined(CONFIG_USB_CDC_ACM_LOG_LEVEL) && CONFIG_USB_CDC_ACM_LOG_LEVEL != LOG_LEVEL_NONE
+#define HAS_USB_CONSOLE DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_console), zephyr_cdc_acm_uart)
+#define HAS_USB_SHELL   DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_shell_uart), zephyr_cdc_acm_uart)
```

Comment:
There is no such thing as USB console or USB shell

```suggestion
#define CHOSEN_CONSOLE DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_console), zephyr_cdc_acm_uart)
#define CHOSEN_SHELL DT_NODE_HAS_COMPAT(DT_CHOSEN(zephyr_shell_uart), zephyr_cdc_acm_uart)
```